src/mixins/labels/base.ts

```diff
@@ -1,87 +1,50 @@
 import * as THREE from "three";
 
 import settings from "../../settings";
-import { ATOM_GROUP_NAME } from "../../enums";
 import { BaseTHREEGroupManager } from "../base";
+import { VerticesHashMapHandler } from "../Hashmap";
 
-export class BaseLabelsManager extends BaseTHREEGroupManager {
-    labelType: string;
+export type LabelsManagerConstructor<T extends BaseLabelsManager> = new (
+    waveStructureGroup: THREE.Group,
+    waveCamera: THREE.Camera,
+    wave: any,
+) => T;
 
-    #THREETexturesCache = {};
+export abstract class BaseLabelsManager extends BaseTHREEGroupManager {
+    labelType = "";
 
-    textProcessor(atom: THREE.Mesh, position: THREE.Vector3) {
-        return atom.name + " " + atom.position.x;
-    }
+    #THREETexturesCache: { [key: string]: THREE.Texture } = {};
```

Comment:
I think it's better to use the `protected` keyword instead of `#`
It has the same meaning, but using `public`, `protected`, and `private` TS keywords is better for consistency  reasons